core/src/main/java/org/pentaho/platform/dataaccess/datasource/api/AnalysisService.java

```diff
@@ -448,12 +452,21 @@ protected IPentahoSession getSession() {
 
   private void fileNameValidation( final String fileName ) throws PlatformImportException {
     if ( fileName == null ) {
-      throw new PlatformImportException( Messages.getString( "AnalysisService.ERROR_001_ANALYSIS_DATASOURCE_ERROR" ),
-          PlatformImportException.PUBLISH_GENERAL_ERROR );
+      throw new PlatformImportException(
+        Messages.getString( "AnalysisService.ERROR_001_ANALYSIS_DATASOURCE_ERROR" ),
+        PlatformImportException.PUBLISH_GENERAL_ERROR
+      );
     }
     if ( fileName.endsWith( METADATA_EXT ) ) {
-      throw new PlatformImportException( Messages.getString( "AnalysisService.ERROR_002_ANALYSIS_DATASOURCE_ERROR" ),
-          PlatformImportException.PUBLISH_GENERAL_ERROR );
+      throw new PlatformImportException(
+        Messages.getString( "AnalysisService.ERROR_002_ANALYSIS_DATASOURCE_ERROR" ),
+        PlatformImportException.PUBLISH_GENERAL_ERROR
+      );
     }
   }
+
+  private boolean shouldListCatalog( MondrianCatalog catalog ) {
+    String toIgnoreProperty = catalog.getConnectProperties().get( "DataAccessNotListedCatalog" );
+    return toIgnoreProperty == null || !toIgnoreProperty.equalsIgnoreCase( "true" );
```

Comment:
Small improvement suggestion: 
```suggestion
    String toIgnoreProperty = catalog.getConnectProperties().get( "DataAccessNotListedCatalog" );
    return !Boolean.parseBoolean(toIgnoreProperty);;
```
Also, the variable name `toIgnoreProperty` seemed confusing here? Can be just me, but wanted to raise the quesiton.